src/xcsync/ProjectFileChangeMonitor.cs

```diff
@@ -67,10 +67,16 @@ public void StartMonitoring (ISyncableProject project, CancellationToken token =
 
 		watcher.EnableRaisingEvents = true;
 
-		var filters = string.Join ("|", this.project.ProjectFilesFilter.Select (f => f.Replace (".", @"\.").Replace ("*", ".*").Replace ("?", ".?")));
-		if (string.IsNullOrEmpty (filters))
-			filters = ".*";
-		fileFilterRegex = new Regex ($"^{filters}$", RegexOptions.IgnoreCase);
+		var filters = this.project
```

Comment:
Instead of handling the extension filtering ourselves, I think the right approach is to use the `Filters` property of the `FileSystemWatcher`. I would even switch the `ProjectFilesFilter` to an array/list to make the initialization easier.

Reply:
There are a few things to consider with the change:

1. how easy is to test the regexpresion? 
2. How error prone is to use an regexpresion?

We can do what @mcumming  mentions and move to Filters. But 'Filters' has a problem, it does not list the files to ignore, but those to watch. 

From the docs:

Filter string | Watches the following files
----------- | ---------------------------
*.* | All files (default). An empty string ("") also watches all files.
*.txt | All files with a "txt" extension.
*recipe.doc | All files ending in "recipe" with a "doc" extension.
win*.xml | All files beginning with "win" with an "xml" extension.
Sales*200?.xls | Matches the following:- Sales July 2001.xls - Sales Aug 2002.xls - Sales March 2004.xlsbut does not match:- Sales Nov 1999.xls
MyReport.Doc | Watches only MyReport.doc

The other option is to create an interface that will represent a filter to pass to the watcher, lets call it IFilesystemEventFilter, that interface can have something like:

```
public interface IFilesystemEventFilter {
    bool ProcessEvent (string path); // you will have to list each event
    bool ProcessRenameEvent (string origin, string destination);
}
```

Then you can create diff filter implementations that can be easily tested. You can have diff filtering approaches, regex, hashset etc...

Reply:
thanks for the insight, we were discussing about how it would be nice to ensure certain extensions were excluded from being monitored, so sounds like the interface approach could be a good fix to try here.

Reply:
We don't need an exclusion list, we only need the inclusion list. And by using the Filters property we eliminate the need for a complicated RegEx, they all become simple pattern matching, e.g "*.csproj", "*.xcassets", etc.